File: src/main/java/io/zeitwert/fm/collaboration/model/base/ObjNoteBase.java

package io.zeitwert.fm.collaboration.model.base;

import io.dddrive.obj.model.base.ObjExtnBase;
import io.dddrive.property.model.EnumProperty;
import io.dddrive.property.model.SimpleProperty;
import io.zeitwert.fm.collaboration.model.ObjNote;
import io.zeitwert.fm.collaboration.model.ObjNoteRepository;
import io.zeitwert.fm.collaboration.model.enums.CodeNoteType;

public abstract class ObjNoteBase extends ObjExtnBase implements ObjNote {

	//@formatter:off
	protected final SimpleProperty<Integer> relatedToId = this.addSimpleProperty("relatedToId", Integer.class);
	protected final EnumProperty<CodeNoteType> noteType = this.addEnumProperty("noteType", CodeNoteType.class);
	protected final SimpleProperty<String> subject = this.addSimpleProperty("subject", String.class);
	protected final SimpleProperty<String> content = this.addSimpleProperty("content", String.class);
	protected final SimpleProperty<Boolean> isPrivate = this.addSimpleProperty("isPrivate", Boolean.class);
	//@formatter:on

	protected ObjNoteBase(ObjNoteRepository repository, Object state) {
		super(repository, state);
	}

	@Override
	public ObjNoteRepository getRepository() {
		return (ObjNoteRepository) super.getRepository();
	}

	@Override
	public void doCalcSearch() {
		this.addSearchText(this.getSubject());
		this.addSearchText(this.getContent());
	}

	@Override
	protected void doCalcAll() {
		super.doCalcAll();
		this.calcCaption();
	}

	private void calcCaption() {
		this.setCaption("Notiz");
	}

}

File: fm-domain/src/main/java/io/zeitwert/fm/collaboration/model/base/ObjNoteBase.kt
package io.zeitwert.fm.collaboration.model.base

import io.dddrive.core.ddd.model.Aggregate
import io.zeitwert.fm.collaboration.model.ObjNote
import io.zeitwert.fm.collaboration.model.ObjNoteRepository
import io.zeitwert.fm.collaboration.model.enums.CodeNoteType
import io.zeitwert.fm.obj.model.base.FMObjBase

abstract class ObjNoteBase(
	override val repository: ObjNoteRepository,
	isNew: Boolean,
) : FMObjBase(repository, isNew),
	ObjNote {

	override fun doInit() {
		super.doInit()
		addBaseProperty("relatedToId", Any::class.java)
		addEnumProperty("noteType", CodeNoteType::class.java)
		addBaseProperty("subject", String::class.java)
		addBaseProperty("content", String::class.java)
		addBaseProperty("isPrivate", Boolean::class.java)
	}

	override fun doCalcAll() {
		super.doCalcAll()
		this.calcCaption()
	}

	private fun calcCaption() {
		setCaption("Notiz: " + (subject ?: "Ohne Betreff"))
	}

	// ObjNote interface implementation

	override val relatedTo: Aggregate?
		get() {
			// TODO: Implement via repository directory lookup
			return null
		}

}
